apps/emqx_retainer/src/emqx_retainer_mnesia.erl

```diff
@@ -706,3 +704,11 @@ are_indices_updated(Indices) ->
         _ ->
             false
     end.
+
+ets_stream(Tab) ->
+    emqx_utils_stream:ets(
+        fun
+            (undefined) -> ets:match_object(Tab, '_', 1);
```

Comment:
I suspect it would be a good idea to see how this rewrite affects performance.

I have a feeling that this approach (iterating one record at a time) will hit it the hardest, however it doesn't seem too hard to adapt it to larger pages though, on the `emqx_utils_stream` side.

Reply:
Yes, that seems worth implementing

Reply:
✅